packages/react-await/src/types.ts

```diff
@@ -0,0 +1,16 @@
+import type { Tuple } from './utility-types'
+
+export type Key = Tuple
+
+/**
+ * @experimental This is experimental feature.
+ */
+export type AwaitOptions<TData, TKey extends Key> = {
+  key: TKey
+  fn: (options: { key: TKey }) => Promise<TData>
+}
+
+export type Awaited<TData> = {
+  data: TData
+  reset: () => void
+}
```

Comment:
I separated it because it depends on a lot of files

Reply:
Cool